internal/eval/eval.go

```diff
@@ -4115,7 +4115,7 @@ func evalJSONSTRLEN(args []string, store *dstore.Store) []byte {
 		jsonData := obj.Value
 
 		if utils.GetJSONFieldType(jsonData) != utils.StringType {
-			return diceerrors.NewErrWithFormattedMessage(diceerrors.JSONPathValueTypeErr)
+			return diceerrors.NewErrWithFormattedMessage(diceerrors.JSONPathValueTypeErr, strings.ToLower(utils.GetJSONFieldType(jsonData)))
```

Comment:
suggestion:
```suggestion
                         currType := strings.ToLower(utils.GetJSONFieldType(jsonData))
                         if currType == "number"{
                                  currType = "integer"
                         }
			return diceerrors.NewErrWithFormattedMessage(diceerrors.JSONPathValueTypeErr, currType)
```

Reply:
Wouldn't it be better to return number for all the float values and integer for all the numbers without decimals instead of returning integer everything.